src/common/dttypes.h

```diff
@@ -49,7 +49,7 @@ typedef float DT_ALIGNED_ARRAY dt_colormatrix_t[4][4];
 // is passed to a function (this kills the compilation due to
 // -Werror).  The most elegant way to handle this is to suppress the
 // warning for the handfull of function calls the bug affects.
-#if __GNUC__ == 12
+#if __GNUC__ >= 12
```

Comment:
Like in other instances recently, we should really try to fix this rather than sweep it under the rug.

Where's the exact error coming from?

Maybe separate it from this ICC PR so we can discuss/analyse it further?

Reply:
> Like in other instances recently, we should really try to fix this rather than sweep it under the rug.

Go ahead, will be a nice fix. Thanks. Meanwhile, those users who build the current git with the gcc 13 can compile it.

> Where's the exact error coming from?

Please see #14143 from @ralfbrown.

> Maybe separate it from this ICC PR so we can discuss/analyse it further?

Why is it needed? How does this small fix interfere with discussing/analyzing this issue further?

Reply:
> Why is it needed?

Good engineering practice.

> How does this small fix interfere with discussing/analyzing this issue further?

It blanketly eliminates all GCC versions >12, which is maybe not what we want as we might forget to fix and remove this workaround in the future. For starters, it would be good to eliminate just the known buggy ones for example.

See, we're already polluting the real and great purpose of this PR w/ unnecessary discussion.

Please separate.

Reply:
Obviously, the sequence should be completely different. I don't see the logic in stopping a quick _temporary_ fix until you've done something. If you already have the correct fix to the code that doesn't trigger a compiler warning, just do a PR. As long as it's not there and you don't guarantee when it will be done, my little fix won't stop you from finding the right solution.

Let's not waste our time arguing, okay?

Reply:
I never said not to commit it, although a different fix is indeed preferable.

I just kindly asked to do things properly and submit it as separate PR that can be more sensibly referenced, reverted if need be, etc.

And I agree, there should be no argument over this.

Reply:
Ah, OK, i just didn't get your point as I didn't get why this small fix, let's say revert (or do anything with it at all). Once you've made the correct fix, you'll simply remove the corresponding disable-warning macro, whether it's in ralfbrown's original form or with this little fix.

Reply:
Correct. Much easier to trace and connect the historical dots through 'git blame' as well for example. Just simple "best practices". Thanks.